src/prefect/engine/task_runner.py

```diff
@@ -879,14 +879,11 @@ def get_task_run_state(self, state: State, inputs: Dict[str, Result]) -> State:
                 new_state.loop_count
             )
 
-        except signals.PrefectStateSignal:
-            # Other state signals will be handled by the `call_state_handlers`
-            # decorator. Once Prefect signal exceptions are modified to inherit
-            # from `BaseException` instead of `Exception`, this can be removed
-            raise
-        except signals.ENDRUN:
-            # As above, the `ENDRUN` is a special exception
-            raise
+        except signals.SUCCESS as exc:
```

Comment:
I think you still need the other special cases to prevent the catch-all handling on line 889, but otherwise this special case definitely makes sense to me

Reply:
Sweet. Since #4664 changes our internal signals to `BaseException` we don't.